pushy/src/main/java/com/turo/pushy/apns/server/RejectionReason.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2013-2017 Turo
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.turo.pushy.apns.server;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * An enumeration of reasons a push notification may be rejected by an APNs server. The most up-to-date descriptions of
+ * each rejection reason are available in Table 8-6 of Apple's <a href="https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html#//apple_ref/doc/uid/TP40008194-CH11-SW1">Local
+ * and Remote Notification Programming Guide - Communicating with APNs</a>.
+ */
+public enum RejectionReason {
```

Comment:
I'm really torn about how much to document in this enumeration. We've learned things that aren't really clear about some of the error codes from the (sparse) official documentation, but I'm hesitant to make strong statements of fact about things that we're still somewhat guessing about and that may change in the future.

I also feel weird about having this enum out in public view, but not using it in `PushNotificationResponse#getRejectionReason()`. The danger in using it there is that if Apple adds a new rejection reason, old versions of Pushy will break when they receive those new codes and can't map them to an enum key. We could have a catch-all `UNKNOWN` key, but that feels pretty gross. We could also add a `getRejectionReasonForErrorCode(String)` type of method, but that seems like so many extra steps that it wouldn't be worth it for most users.

Very open to ideas.